clang/tools/libclang/CCAS.cpp

```diff
@@ -13,12 +13,68 @@
 
 #include "clang/Basic/LLVM.h"
 #include "clang/CAS/CASOptions.h"
+#include "clang/Frontend/TextDiagnosticPrinter.h"
 #include "llvm/CAS/ActionCache.h"
 #include "llvm/CAS/ObjectStore.h"
+#include "llvm/Support/Path.h"
 
 using namespace clang;
 using namespace clang::cas;
 
+CXCASOptions clang_experimental_cas_Options_create(void) {
+  return wrap(new WrappedCASOptions());
+}
+
+void clang_experimental_cas_Options_dispose(CXCASOptions Opts) {
+  delete unwrap(Opts);
+}
+
+void clang_experimental_cas_Options_setOnDiskPath(CXCASOptions COpts,
+                                                  const char *Path) {
+  CASOptions &Opts = unwrap(COpts)->CASOpts;
+  SmallString<256> PathBuf;
+  PathBuf = Path;
+  llvm::sys::path::append(PathBuf, "cas");
+  Opts.CASPath = std::string(PathBuf);
+  PathBuf = Path;
```

Comment:
You can drop the last component instead of re-copying the whole string

```
llvm::sys::path::remove_filename(PathBuf);
```

Reply:
Done.